### ISSUE TITLE
kjj

### ISSUE THREAD

**Issue by muddham** [tensorflow/tensorflow#28800]
import tensorflow as tf
from tensorflow.python import keras
from tensorflow.python.keras import layers as KL
from tensorflow.python.keras import models as KM
import numpy as np
print(tf.__version__)

def my_custom_loss_wrapper(input_weight):
    def real_loss(y_true, y_pred):
        # expand the out put of binary_crossentropy(64,64) to (64,64,1) to match the shape of input_weight
        bce_loss = tf.expand_dims(keras.losses.binary_crossentropy(y_true, y_pred), axis=-1)
        return bce_loss * input_weight

    return real_loss

fake_img = np.ones((64, 64, 3))
fake_label = np.ones((64, 64, 1))
fake_weight = np.ones((64, 64, 1)) * 5
dataset = tf.data.Dataset.from_tensors(((fake_img, fake_weight), fake_label)).repeat(100).batch(10)

img_input = KL.Input(shape=(64, 64, 3))
weight_input = KL.Input(shape=(64, 64, 1))
x = KL.Conv2D(32, (3, 3), strides=2, padding="same")(img_input)
x = KL.Conv2DTranspose(32, (3, 3), strides=2, padding="same")(x)
mask = KL.Conv2D(1, (1, 1), strides=1, activation="sigmoid", name="mask")(x)
model = KM.Model(inputs=[img_input, weight_input], outputs=mask)
model.compile(
    loss=my_custom_loss_wrapper(weight_input),  # return a function real_loss(y_true, y_pred)
    optimizer='sgd'
)
model.fit(dataset, epochs=1000)

strategy = tf.distribute.MirroredStrategy(
    cross_device_ops=tf.distribute.HierarchicalCopyAllReduce()
)
with strategy.scope():
    img_input = KL.Input(shape=(64, 64, 3))
    weight_input = KL.Input(shape=(64, 64, 1))
    x = KL.Conv2D(32, (3, 3), strides=2, padding="same")(img_input)
    x = KL.Conv2DTranspose(32, (3, 3), strides=2, padding="same")(x)
    mask = KL.Conv2D(1, (1, 1), strides=1, activation="sigmoid", name="mask")(x)
    model = KM.Model(inputs=[img_input, weight_input], outputs=mask)
    model.compile(
        # loss="mse",
        loss=my_custom_loss_wrapper(weight_input),  # return a function real_loss(y_true, y_pred)
        optimizer='sgd'
    )
model.fit(dataset, epochs=1000)

**Comment by gadagashwini** [comment]
`def model():
    model_input = Input(shape=input_shape[1:], name="input_layer")
    x = Conv2D(64, (4, 30), padding="same", name="layer1a")(model_input)
    x = BatchNormalization()(x)
    x = Activation("relu")(x)
    x = MaxPool2D()(x)

    x = Conv2D(64, (4, 30), padding="same")(x)
    x = BatchNormalization()(x)
    x = Activation("relu")(x)
    x = MaxPool2D()(x)`
